Orange/widgets/utils/saveplot.py

```diff
@@ -0,0 +1,42 @@
+import os.path
+from operator import attrgetter
+
+from PyQt4 import QtGui, QtCore
+
+
+# noinspection PyBroadException
+def save_plot(data, file_formats, filename=""):
+    formats = [(f.DESCRIPTION, f.EXTENSIONS)
+               for f in sorted(set(file_formats.values()),
+                               key=attrgetter("OWSAVE_PRIORITY"))]
+    filters = ['{} (*{})'.format(desc, ' *'.join(exts))
+               for desc, exts in formats]
+
+    _LAST_DIR_KEY = "directories/last_graph_directory"
+    _LAST_EXT_KEY = "directories/last_graph_extension"
+    settings = QtCore.QSettings()
+    start_dir = settings.value(_LAST_DIR_KEY, filename)
+    if not start_dir or not os.path.exists(start_dir):
+        start_dir = os.path.expanduser("~")
+    last_ext = settings.value(_LAST_EXT_KEY, "")
+    if last_ext not in filters:
+        last_ext = filters[0]
+
+    filename, filter = QtGui.QFileDialog.getSaveFileNameAndFilter(
+        None, 'Save as ...', start_dir, ';;'.join(filters), last_ext)
+    if not filename:
+        return
+
+    ext = os.path.splitext(filename)[1]
+    exts = formats[filters.index(filter)][1]
```

Comment:
This is what would happen on Windows unless you put the name in quotes. But I agree, this is annoying. Also, on Os X the system dialog warns you that you're specifying a wrong extension and asks you to confirm -- and after confirmation this function adds the unwanted extension anyway. :)

What do you suggest? I guess we add an extension only if `filename` has none, that is, if there is no dot after slash?